### PR TITLE
Add Bracket Pair Colorization for Improved Readability

### PR DESCRIPTION
**Pull request by morgilad**
**Changes:**

- **Bracket Pair Colorization:**  
  Added highlighting for matching bracket pairs to improve code readability.

**Testing:**

- Verified bracket highlighting works as expected.

### CODE REVIEW

File: package/tests/integration/editor.spec.ts [hunk 1]
```diff
@@ -36,5 +36,15 @@ test.describe('editor', () => {
...
         });
+
+        test('should highlight matching pairs', async ({ page }) => {
+            const bracketElements = page.locator('.bracket-match');
+
+            await page.keyboard.type('StormEvents\n' + '| where (datetime() < ago(1h))\n');
+            await expect(bracketElements).not.toBeVisible();
+
+            await page.keyboard.press('ArrowLeft');
+            await expect(bracketElements).toHaveCount(2);
```
Comment (izikl) [inline package/tests/integration/editor.spec.ts:47]: is this testing highlighting or the auto-addition of a closing bracket?

**Review by izikl — APPROVED** [review]
I will trust you that it is working. Didn't verify.

File: package/tests/integration/editor.spec.ts (hunk 1 above)
Comment (morgilad) [reply package/tests/integration/editor.spec.ts:47]: highlighting. auto addition is tested in the tests above

Comment (morgilad) [reply package/tests/integration/editor.spec.ts:47]: bracket-match is added when the highlighting of matching pairs is enabled

**Pull request merged**